Lib/_collections_abc.py

```diff
@@ -409,6 +418,94 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _PosArgsGenericAlias(GenericAlias):
+    """ Internal class specifically to represent positional arguments in
+    ``_CallableGenericAlias``.
+    """
+    def __repr__(self):
+        return f"{__name__}._PosArgsGenericAlias" \
+               f"[{', '.join(_type_repr(t) for t in self.__args__)}]"
+
+    def __eq__(self, other):
+        o_cls = other.__class__
+        if not (o_cls.__module__ == "typing" and o_cls.__name__
+                == "_GenericAlias" or isinstance(other, GenericAlias)):
+            return NotImplemented
+        return (self.__origin__ == other.__origin__
+                and self.__args__ == other.__args__)
+
+    def __hash__(self):
+        return hash((self.__origin__, self.__args__))
+
+# Only used for _CallableGenericAlias
+class _PosArgs:
+    """ Internal class specifically to represent positional arguments in
+    ``_CallableGenericAlias``.
+    """
+    def __class_getitem__(cls, item):
+        return _PosArgsGenericAlias(tuple, item)
+
+# _PosArgs = type("_PosArgs", (tuple, ), {})
+
+class _CallableGenericAlias(GenericAlias):
+    """ Internal class specifically for consistency between the ``__args__`` of
+    ``collections.abc.Callable``'s and ``typing.Callable``'s ``GenericAlias``.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if not isinstance(args, tuple) or len(args) != 2:
+            raise TypeError("Callable must be used as Callable[[arg, ...], result]")
+        origin, _args = args
+        if not isinstance(_args, tuple) or len(_args) != 2:
+            raise TypeError("Callable must be used as Callable[[arg, ...], result]")
+        t_args, t_result = _args
+        if not isinstance(t_args, (list, EllipsisType)):
+            raise TypeError("Callable[args, result]: args must be a list "
+                            f"or Ellipsis. Got {_type_repr(t_args)}")
+
+        ga_args = (_args if t_args is Ellipsis
+                   else (_PosArgs[tuple(t_args)], t_result))
```

Comment:
Expand to if-statement? As soon as a conditional expression doesn't fit on a line it becomes awkward IMO. Also below.